state/session/service.js

```diff
@@ -1,27 +1,46 @@
 import { createLogic } from 'redux-logic';
+import cookie from 'cookie_js';
 
 import { API_KEY } from '../../constants';
-import { SESSION_REQUEST_TOKEN } from './types';
-import { sessionRequestTokenSuccess } from './actions';
+import { REQUEST_SESSION } from './types';
+import { requestSessionSuccess } from './actions';
 
-const sessionRequestTokenLogic = createLogic({
-  type: SESSION_REQUEST_TOKEN,
+/* eslint-disable camelcase */
+const requestSessionLogic = createLogic({
+  type: REQUEST_SESSION,
   warnTimeout: 0,
 
-  process({ httpClient }, dispatch, done) {
-    httpClient.get(`authentication/token/new?api_key=${API_KEY}`)
-      .then(
-        (payload) => {
-          const token = payload.data.request_token;
-          dispatch(sessionRequestTokenSuccess(token));
-        }
-      )
-      .catch(
-        /* eslint-disable no-console */
-        (err) => console.error('ERROR', err)
-      )
-      .then(() => done());
+  async process({ httpClient, action }, dispatch, done) {
+    const {
+      username,
+      password,
+      setErrors,
+      setSubmitting
+    } = action;
+
+    try {
+      const { data: responseToken } = await httpClient.get(`authentication/token/new?api_key=${API_KEY}`);
```

Comment:
Move api_key to `httpClient` initialisation to avoid duplication in each API request

Reply:
Done